### PR TITLE
enhancement: add error handling during login

### LINKED ISSUE #3918: [Task]: Handle error for incorrect device time
### Task description

Some users are suffering from a bug that prevents their profiles from accessing the wallet view. There application hangs on the Firefly loading screen, and they continuously see the Firefly logo's loading animation. Restarting the application and logging into the profile also produces the same issue.

OS's reporting this error:
- Windows 10
- Windows 11

I am able to reproduce this behavior on my Windows 11 laptop.

### Requirements

Capture error that causes this freeze by setting up a new specialized Firefly alpha/beta version that enables developer console.

### Creation checklist

- [X] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### LINKED ISSUE #4123: [Task]: Add error handling during login
### Task description

We should improve error handling during login. And if the user can not login they should be logged out and explained why. IF a profile is corrupt or unrecoverable we could even give them an opportunity to delete it from the login menu.

### Requirements

TBD

### Creation checklist

- [ ] I have assigned this task to the correct people
- [x] I have added the most appropriate labels
- [x] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

Improve core error module to include handling errors and use the new functionality to add error handling to the login function.

## Changelog

```
- Make sure errors are bubbling up to login() catch block
- Create handleError functions in core/error/utils
- Add constants and enums to core/error to support errorHandling
```

## Relevant Issues

> Please list any related issues using [development keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Closes #4123 and Closes #3918

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [x] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/error/constants/api-error-parameters.constant.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { ClientError, ErrorFromApi } from '../enums'
+import { IErrorParameters } from '../interfaces'
+
+export const API_ERROR_PARAMETERS: Readonly<
+    { [key in ErrorFromApi]?: { [key in ClientError]?: Partial<IErrorParameters> } }
```
Comment (MarkNerdi) [inline packages/shared/lib/core/error/constants/api-error-parameters.constant.ts:5]: i think this type isn't correct.
If the toplevel key is `ErrorFromApi.ClientError` then the subkeys can be of type `ClientError`.
But if the toplevel key is e.g. `ErrorFromApi.InsufficientFundsError`  (which you get from walletrs as `{type: "InsufficientFunds", message: "insufficient amount of funds ..."}`, we wouldn't add these subtypes in the `ClientError` enum

File: packages/shared/lib/core/error/utils/handleClientError.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { API_ERROR_PARAMETERS, CLIENT_ERROR_REGEXES } from '../constants'
+import { ClientError } from '../enums'
+import { IError } from '../interfaces'
+import { handleError } from './handleError'
+import { handleGenericError } from './handleGenericError'
+
+export function handleClientError(error: IError): void {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/error/utils/handleClientError.ts:7]: i would generalize this. i think we would do the exact same for not only errors with `type: ClientError` right?

**Review by MarkNerdi — CHANGES_REQUESTED** [review]
In general it's a nice way to handle error

But the namings are only adapted to Clienterrors, even though we would do the same things for non-`ClientErrors` thrown by wallet.rs. It wouldn't really make sense to create different Enums and Regexs for different `error.type`s. 

Therefore I think we can remove the layer `ErrorFromApi`, because what interests us essentially is only the error message we get.

So 
CLIENT_ERROR_REGEX -> ERROR_REGEX 
ClientError -> (Api?)Error

wdyt?

**Comment by nicole-obrien** [comment]
@MarkNerdi there are a whole load of error types that can be thrown by wallet.rs. Some of them we will just have a catch all error for some we want to check the error message to see if it's a specific error we want to handle in a certain way. Because the switch statements for the regexes would potentially evaluate them all I think it's more efficient and sensible to only do the regexes for the types and messages we want. Rather than creating one linear list. Hence it's probably more efficient to filter on the type first. That's also because the fall back handling for each error type might be different. We might want to show error notification for client errors but not another type etc

**Review by MarkNerdi — APPROVED** [review]

**Pull request merged**